streams/buffer.ts

```diff
@@ -232,3 +232,44 @@ export class LimitedTransformStream<T> extends TransformStream<T, T> {
     });
   }
 }
+
+/**
+ * A transform stream that only transforms from the zero-indexed `start` and `end` bytes (both inclusive).
+ *
+ * @example
+ * ```ts
+ * import { ByteSliceStream } from "https://deno.land/std@$STD_VERSION/streams/buffer.ts";
+ * const response = await fetch("https://example.com");
+ * const rangedStream = response.body!
+ *   .pipeThrough(new ByteSliceStream(3, 8));
+ * ```
+ */
+export class ByteSliceStream extends TransformStream<Uint8Array, Uint8Array> {
+  #offsetStart = 0;
+  #offsetEnd = 0;
+
+  constructor(start = 0, end = Infinity) {
+    super({
+      start: () => {
+        assert(start >= 0, "`start` must be greater than 0");
+        end += 1;
+      },
+      transform: (chunk, controller) => {
+        this.#offsetStart = this.#offsetEnd;
+        this.#offsetEnd += chunk.byteLength;
+        if (this.#offsetEnd > start) {
+          if (this.#offsetStart < start) {
+            chunk = chunk.slice(start - this.#offsetStart);
+          }
```

Comment:
I think we can reduce the number of `slice` calls (which is expensive) if we store `start - this.#offsetStart` value and use it below.

Reply:
Both `slice`s are called only in a rare case - when both `start` and `end` lie just outside the chunk for a given iteration. I think the current code strikes a good balance between clarity and efficiency. What do you think?

Reply:
hmm ok. Now this makes sense to me.